go-runtime/schema/analyzers/verb.go

```diff
@@ -0,0 +1,188 @@
+	"github.com/TBD54566975/golang-tools/go/analysis"
+	"github.com/TBD54566975/golang-tools/go/analysis/passes/inspect"
+	"github.com/TBD54566975/golang-tools/go/ast/inspector"
+	"github.com/alecthomas/types/optional"
+)
+
+// VerbExtractor extracts verbs to the module schema.
+var VerbExtractor = newExtractor("verb", extractVerbs, (*verbExtractorFact)(nil))
+
+type verbExtractorFact struct {
+	decl schema.Decl
+}
+
+func (t *verbExtractorFact) AFact() {}
+func (t *verbExtractorFact) setValue(decl schema.Decl) {
+	t.decl = decl
+}
+func (t *verbExtractorFact) getValue() schema.Decl {
+	return t.decl
+}
+
+func extractVerbs(pass *analysis.Pass) (interface{}, error) {
+	nn := NativeNames{}
+	verbs := []schema.Decl{}
+	in := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector) //nolint:forcetypeassert
+	nodeFilter := []ast.Node{
+		(*ast.FuncDecl)(nil),
+	}
+	in.Preorder(nodeFilter, func(n ast.Node) {
+		node := n.(*ast.FuncDecl) //nolint:forcetypeassert
+		directives := parseDirectives(pass, node, node.Doc)
+		verbDirective, ok := getDirective[*directiveVerb](directives).Get()
+		if !ok {
+			return
+		}
+
+		isExported := verbDirective.IsExported()
+		fnt := pass.TypesInfo.Defs[node.Name].(*types.Func) //nolint:forcetypeassert
+		sig := fnt.Type().(*types.Signature)                //nolint:forcetypeassert
+		if sig.Recv() != nil {
+			pass.Report(errorf(node, "ftl:verb cannot be a method"))
+			return
+		}
+		params := sig.Params()
+		results := sig.Results()
+		reqt, respt := checkSignature(pass, node, sig)
+
+		req := optional.Some[schema.Type](&schema.Unit{})
+		if reqt.Ok() {
+			req = extractType(pass, node.Pos(), params.At(1).Type(), isExported)
+		}
+
+		resp := optional.Some[schema.Type](&schema.Unit{})
+		if respt.Ok() {
+			resp = extractType(pass, node.Pos(), results.At(0).Type(), isExported)
+		}
+
+		reqV, reqOk := req.Get()
+		resV, respOk := resp.Get()
+		if !reqOk {
+			pass.Report(tokenErrorf(pass.Fset, params.At(1).Pos(), params.At(1).Name(),
+				"unsupported request type %q", params.At(1).Type()))
+		}
+		if !respOk {
+			pass.Report(tokenErrorf(pass.Fset, results.At(0).Pos(), results.At(0).Name(),
+				"unsupported response type %q", results.At(0).Type()))
+		}
+
+		metadata := extractVerbMetadata(pass, node, directives)
+		for _, m := range metadata {
+			switch m.(type) {
+			case *schema.MetadataIngress:
+				isExported = true
+			case *schema.MetadataCronJob:
+				isExported = false
+			default:
+			}
+		}
+
+		verb := &schema.Verb{
+			Pos:      goPosToSchemaPos(pass.Fset, node.Pos()),
+			Comments: extractComments(node.Doc),
+			Export:   isExported,
+			Name:     strcase.ToLowerCamel(node.Name.Name),
+			Request:  reqV,
+			Response: resV,
+			Metadata: metadata,
+		}
+		nn[verb] = node.Name.Name
+		verbs = append(verbs, verb)
+	})
+
+	return newExtractorResult(pass, verbs, nn), nil
+}
+
+func extractVerbMetadata(pass *analysis.Pass, node ast.Node, directives []directive) []schema.Metadata {
```

Comment:
Metadata extraction could be another pass. We could look for any `//ftl:` directives on any declaration, and add them as facts to the node. Then in verb analysis, pubsub, types, etc. we just pull the metadata fact directly from the node.